skglm/solvers/base.py

```diff
@@ -0,0 +1,41 @@
+from abc import abstractmethod
+
+
+class BaseSolver():
+    """Base class for solvers."""
+
+    @abstractmethod
+    def solve(self, X, y, datafit, penalty, w_init, Xw_init):
+        """Solve an optimization problem.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            Training data.
+
+        y : array, shape (n_samples,)
+            Target values.
+
+        datafit : instance of Datafit class
+            Datafitting term.
+
+        penalty : instance of Penalty class
+            Penalty used in the model.
+
+        w_init : array, shape (n_features,)
+            Coefficient vector.
```

Comment:
```suggestion
            Initial coefficient vector. It is used by the solver as an initial guess of the solution.
```

Reply:
see commend above: I agree on the principle, but I'm not in favor of doing it here